### PR TITLE
[all] Clean unused warnings in SOFA

### PR DESCRIPTION
**Pull request by hugtalbot**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/PrecomputedConstraintCorrection.cpp [hunk 1]
```diff
@@ -42,7 +42,6 @@ SOFA_COMPONENT_CONSTRAINT_LAGRANGIAN_CORRECTION_API void PrecomputedConstraintCo
 
     for (auto rowIt = c.begin(); rowIt != rowItEnd; ++rowIt)
     {
-        auto rowWrite = c.writeLine(rowIt.index());
```
Comment (damienmarchal) [inline Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/PrecomputedConstraintCorrection.cpp:45]: Beware here... To me the warning indicates that rowWrite is not used... not that writeLine is useless.

File: Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/PrecomputedConstraintCorrection.inl [hunk 2]
```diff
@@ -873,7 +873,6 @@ void PrecomputedConstraintCorrection< DataTypes >::rotateConstraints(bool back)
 
     for (auto rowIt = c.begin(); rowIt != rowItEnd; ++rowIt)
     {
-        auto rowWrite = c.writeLine(rowIt.index());
```
Comment (damienmarchal) [inline Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/PrecomputedConstraintCorrection.inl:876]: Beware here... the warning indicates that rowWrite is not used... not that writeLine is useless. Unless there is proves that writeLine cannot do anything then we need to keep it.

Comment (fredroy) [reply Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/PrecomputedConstraintCorrection.inl:876]: I agree ; nevertheless I think we can remove the warning with [[maybe_unused]]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Review by fredroy — APPROVED** [review]

**Pull request merged**